metron-deployment/packaging/ambari/README.md

```diff
@@ -0,0 +1,86 @@
+  ```
+  new_property = status_params.new_property
+  ```
+This behavior is because Ambari doesn't send all parameters to the status, so it needs to be explicitly provided.
+
+3. Ambari master services can then import the params as such
+
+  ```
+  from params import params
+  env.set_params(params)
+  ```
+
+4. The `*_commands.py` files receive the params as an input from the master services.  Once this is done, they can be accessed via the variable we set above:
+  ```
+  self.__params.new_property
+  ```
+
+  Ambari also has a custom `format` function that can access properties when producing strings as output:
+  ```
+  from resource_management.libraries.functions import format as ambari_format
+  # ...
+  Foo(ambari_format("{new_property}-adjusted")),
+  ```
+
+## How to identify errors in MPack changes
+Typically, any errors are thrown at one of two times:
+
+1. Attempting to install Metron as a whole.  These are typically service level definition errors, not property errors. Logs will often be found in `/var/log/ambari-server/ambari-server.log`.  Often the log will indicate it was unable to load a particular service or subservice with some more detail.
+2. Running the actual functionality.  These typically tend to cause errors in the UI at runtime.  Logs are usually found in `/var/log/ambari-agent/ambari-agent.log`.
+
+Unfortunately, because errors tend to occur at runtime, it's often necessary to add things like logging statements, or even just throw errors to print out in the Ambari UI.
+
+The primary solution to these is to look in the logs for exceptions, see what's going wrong (Property doesn't exist?  Malformed file couldn't be loaded?), and adjust appropriately.
+
+## Testing changes without cycling Vagrant build
+To avoid spinning down and spinning back up Vagrant, we'll want to instead modify the files Ambari uses directly.
+Ambari stores the Python files from the service in a couple places.
+
+Specifically, the server files live in
+```
+/var/lib/ambari-server/resources/mpacks/metron-ambari.mpack-0.4.0.0/common-services
+/var/lib/ambari-agent/cache/common-services
+```
+
+e.g. enrichment_commands.py can be found in:
+
+```
+/var/lib/ambari-server/resources/mpacks/metron-ambari.mpack-0.4.0.0/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+/var/lib/ambari-agent/cache/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+```
+
+A `find` command can also be useful in quickly locating the exact location of a file, e.g.
+```
+[root@node1 ~]# find /var/lib/ -name enrichment_commands.py
+/var/lib/ambari-server/resources/mpacks/metron-ambari.mpack-0.4.0.0/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+/var/lib/ambari-agent/cache/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+```
+
+The steps to update are:
+
+1. Edit the file(s) with your changes.  The ambari-agent file must be edited, but generally better to update both for consistency.
```

Comment:
This relates to the gap @merrimanr is looking at about the redeployment of the mpack (which addresses the first situation).  This hits the second situation.  Once I get the proper set of steps (since I'm pretty sure I'll miss something if I do it), I'll update this to be explicit about what each is addressing.